ui-v2/app/components/modal-dialog.js

```diff
@@ -0,0 +1,119 @@
+import { get, set } from '@ember/object';
+import { inject as service } from '@ember/service';
+import Component from 'consul-ui/components/dom-buffer';
+import SlotsMixin from 'ember-block-slots';
+import WithResizing from 'consul-ui/mixins/with-resizing';
+
+import templatize from 'consul-ui/utils/templatize';
+export default Component.extend(SlotsMixin, WithResizing, {
+  dom: service('dom'),
+  checked: true,
+  height: null,
+  // dialog is a reference to the modal-dialog 'panel' so its 'window'
+  dialog: null,
+  overflowingClass: 'overflowing',
+  onclose: function() {},
+  onopen: function() {},
+  _open: function(e) {
+    set(this, 'checked', true);
+    if (get(this, 'height') === null) {
+      if (this.element) {
+        const dialogPanel = get(this, 'dom').element('[role="dialog"] > div > div', this.element);
```

Comment:
Thanks for the suggestion here, I can partly see where you are coming from, but...

I suppose this is similar to the conversation we had around CSS selectors and HTML/template structure.

Usually if I need to change one, it is absolutely no problem to change the other. Indeed in this case other frameworks like React and Vue etc let you, even encourage you, to keep your template in the exact same file as your javascript behavior (that's not something I'm keen on myself).

As far as I'm aware with the ember module unification effort, all of these files will live in the same place, coupling them even more and making them even easier to make changes like this.

> I'd feel a little better if it were an id

`id`s should be unique. With this component it's probably not the case, but if I were to add a component to the same page twice, and the component template used an `id` in it, it could produce one of those hard to track down bugs. So as a general rule I would almost never put an `id` in a component.